modules/seqexec/server/src/main/scala/seqexec/server/gems/Gems.scala

```diff
@@ -31,12 +38,35 @@ object Gems {
     override def pauseResume(config: GemsConfig,
                              pauseReasons: Set[Gaos.PauseCondition],
                              resumeReasons: Set[ResumeCondition]): F[PauseResume[F]] = {
-      Applicative[F].pure(PauseResume(None, None))
+      controller.pauseResume(cfg, pauseReasons, resumeReasons)
+    }
+
+    override def usesP1(guide: GemsConfig): Boolean = guide match {
+      case GemsOn(_, _, _, _, _, _, _, _, useP1) => useP1
+      case _                                     => false
+    }
+
+    override def usesOI(guide: GemsConfig): Boolean = guide match {
+      case GemsOn(_, _, _, _, _, _, _, useOI, _) => useOI
+      case _                                     => false
     }
+
+    override val stateGetter: GemsGuiderStatus[F] = controller.stateGetter
   }
 
   def fromConfig[F[_]: Applicative](controller: GemsController[F])/*(config: Config)*/: F[Gems[F]] =
     Applicative[F].pure(new GemsImpl[F](controller, GemsOff))
+
+  final case class GemsGuiderStatus[F[_]](
+    ngs1: F[Option[Boolean]],
+    ngs2: F[Option[Boolean]],
+    ngs3: F[Option[Boolean]],
+    odgw1: F[Option[Boolean]],
+    odgw2: F[Option[Boolean]],
+    odgw3: F[Option[Boolean]],
+    odgw4: F[Option[Boolean]]
+  )
```

Comment:
This class is used to retrieve the GeMS guiders active state (i.e. if they are producing error measurements or not). They were put in a case class so the implementation can be switched between the simulated and real version.

Reply:
Those values have a different meaning. The types I created before are used to indicate a request to use a specific guider for GeMS. The values in `GemsGuiderStatus` are the procedures used to know if a guider is currently on. Actually, that value does not reflect the whole guider state, only if it is producing measurements (the detector could be reading out, but the probe may not be following the target, making the measurements useless). I see that the names don't match, I'll fix them.